### PR TITLE
Add machine reading comprehension model, preprocessor and pipeline

### PR DESCRIPTION
**Pull request by RandyZhouRan**
This pull request aims to add support for a new NLP task: machine reading comprehension (MRC)
Changes include:
1. Add a new task model ``ModelForMachineReadingComprehension''
2. Add preprocessor and pipeline for the ``Pretrained Machine Reader (PMR) for NER''
    which is from the paper [From Cloze to Comprehension: Retrofitting Pre-trained Language Models to Pre-trained Machine Readers](https://arxiv.org/pdf/2212.04755.pdf)

### CODE REVIEW

**Comment by RandyZhouRan** [comment]
The unit test is created as `tests/pipelines/test_machine_reading_comprehension.py`, and is cleared when running locally

File: modelscope/preprocessors/nlp/machine_reading_comprehension_preprocessor.py [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+
+        if label2query is None:
+            config_path = os.path.join(model_dir, ModelFile.CONFIGURATION)
+            config = Config.from_file(config_path)
+            self.label2query = config[ConfigFields.preprocessor].label2query
+        else:
+            self.label2query = label2query
+
+    def __call__(self, data: str):
+        all_data = []
```
Comment (wenmengzhou) [inline modelscope/preprocessors/nlp/machine_reading_comprehension_preprocessor.py:44]: 直接使用tokenizer做decode？   必须要构建Dataset和DataLoader

File: modelscope/pipelines/nlp/machine_reading_comprehension_pipeline.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+from modelscope.preprocessors import Preprocessor
+from modelscope.utils.constant import Fields, Tasks
+
+
+@PIPELINES.register_module(
+    Tasks.machine_reading_comprehension,
+    module_name=Pipelines.machine_reading_comprehension_for_ner)
+class MachineReadingComprehensionForNERPipeline(Pipeline):
+    '''
+    Pipeline for Pretrained Machine Reader (PMR) finetuned on Named Entity Recognition (NER)
```
Comment (wenmengzhou) [inline modelscope/pipelines/nlp/machine_reading_comprehension_pipeline.py:19]: 参考 https://github.com/modelscope/modelscope/blob/master/modelscope/pipelines/cv/card_detection_pipeline.py#L18补充pipeline注释

File: modelscope/pipelines/nlp/machine_reading_comprehension_pipeline.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        with torch.no_grad():
+            outputs = self.model(**inputs)
+        span_logits = outputs['span_logits']
+
+        return MachineReadingComprehensionOutput(
+            span_logits=span_logits,
+            input_ids=inputs['input_ids'],
+        )
+
+    def postprocess(
```
Comment (wenmengzhou) [inline modelscope/pipelines/nlp/machine_reading_comprehension_pipeline.py:59]: pipeline对应的任务类型需要有输入输出字段名称的定义
      输入请在modelscope/pipeline_inputs.py中补充
      输出请在 modelscope/outputs/outputs.py  补充

File: modelscope/preprocessors/nlp/machine_reading_comprehension_preprocessor.py (hunk 1 above)
Comment (RandyZhouRan) [reply modelscope/preprocessors/nlp/machine_reading_comprehension_preprocessor.py:44]: 已优化，保留数据处理逻辑，省去Dataset和Dataloader

File: modelscope/pipelines/nlp/machine_reading_comprehension_pipeline.py (hunk 3 above)
Comment (RandyZhouRan) [reply modelscope/pipelines/nlp/machine_reading_comprehension_pipeline.py:67]: 已添加

**Review by wenmengzhou — APPROVED** [review]

**Pull request merged**